src/test/java/org/commcare/formplayer/tests/RestoreFactoryTest.java

```diff
@@ -311,6 +315,23 @@ public void testGetRequestHeaders_UseHmacAuthEvenIfHqAuthPresent() throws Except
         );
     }
 
+    @Test
+    public void testChecksForLocationChanges() throws Exception {
```

Comment:
can we also do a check for multiple locations when unchanged, I want to verify hasLocationChanged remains `false` when same multiple locations are assigned to user before and after but in a a different order of array items.

Reply:
Actually, it turns out when you assign a user multiple locations, the order you choose them in the location selector really doesn't matter, so this test is unnecessary. This is something I've noticed before (which is why I had it check the whole string) but it's apparently because django's multiple selector widgets (we specifically use our own `LocationSelectWidget` but I believe it applies here, since I'm seeing the same behavior) will always order user selected options in the order they appear in the initial choice dict.

Reply:
@minhaminha I don't think Formplayer should rely on another system to send it locations in a particular order. The question here is more -  if the locations are indeed in a random order, will Formplayer do the right thing here or not ?